frontend/javascripts/oxalis/model/volumetracing/volumelayer.js

```diff
@@ -238,6 +239,130 @@ class VolumeLayer {
     return iterator;
   }
 
+  perpendicularVector2(pos1: Vector2, pos2: Vector2): Vector2 {
+    const dx = pos2[0] - pos1[0];
+    let perpendicularVector;
+    if (dx === 0) {
+      perpendicularVector = [1, 0];
+    } else {
+      const gradient = (pos2[1] - pos1[1]) / dx;
+      perpendicularVector = [gradient, -1];
+      const norm = this.normOfVector2(perpendicularVector);
+      perpendicularVector = this.scalarMultiplication(perpendicularVector, 1 / norm);
+    }
+    return perpendicularVector;
+  }
+
+  normOfVector2(vector: Vector2): number {
+    let norm = 0;
+    let i;
+    for (i of Vector2Indicies) {
+      norm += Math.pow(vector[i], 2);
+    }
+    return Math.sqrt(norm);
+  }
+
+  vector2Sum(vector1: Vector2, vector2: Vector2): Vector2 {
+    const sum = [0, 0];
+    let i;
+    for (i of Vector2Indicies) {
+      sum[i] = vector1[i] + vector2[i];
+    }
+    return sum;
+  }
+
+  scalarMultiplication(vector: Vector2, scalar: number): Vector2 {
+    const product = [0, 0];
+    let i;
+    for (i of Vector2Indicies) {
+      product[i] = vector[i] * scalar;
+    }
+    return product;
+  }
+
+  vector2Distance(pos1: Vector2, pos2: Vector2): number {
+    let distance = 0;
+    let i;
+    for (i of Vector2Indicies) {
+      distance += Math.pow(pos2[i] - pos1[i], 2);
+    }
+    return Math.sqrt(distance);
+  }
+
+  vectorWithScale(vector: Vector2, scale: Vector2): Vector2 {
```

Comment:
As all these new methods are for Vector2, I would suggest to use a more similar naming pattern for all of them and let all of the method names include the term `vector2`.